Automattic/woocommerce-payments
# Add welcome tour component

Fixes Automattic/woocommerce-payments-server#2786

#### Changes proposed in this Pull Request

Add the welcome tour components

This component provides a tour experience based on the configuration steps passed as parameters and should work in a generic way.

Figma reference (WIM5w2Y10SCNBFdMLSVftj-fi-1300%3A196843)









https://user-images.githubusercontent.com/16882226/223614496-1a57dc5a-0a6a-41f1-bcb6-45f416dfa270.mp4

#### Testing instructions





* Checkout to this branch `add/2786-welcome-tour`
* Run `npm start`
* Apply the "Example diff" below
* Go to **Payments > Settings**
* The tour modal should pop up
  * Navigate through it
  * You can close the modal by pressing the "X" button or clicking on the "Got it" button at the end
  * Try to resize the browser window
  * It should be centered on mobile

<details>
  <summary>Example diff</summary>

  ```diff
diff --git a/client/settings/settings-manager/index.js b/client/settings/settings-manager/index.js
index d6e0878b..43cb26c1 100644
--- a/client/settings/settings-manager/index.js
+++ b/client/settings/settings-manager/index.js
@@ -2,7 +2,7 @@
 /**
  * External dependencies
  */
-import React, { useContext, useState } from 'react';
+import React, { useContext, useState, useEffect } from 'react';
 import { ExternalLink } from '@wordpress/components';
 import { __, sprintf } from '@wordpress/i18n';
 
@@ -22,7 +22,8 @@ import WCPaySettingsContext from '../wcpay-settings-context';
 import LoadableSettingsSection from '../loadable-settings-section';
 import WcPayUpeContextProvider from '../wcpay-upe-toggle/provider';
 import ErrorBoundary from '../../components/error-boundary';
-import { useDepositDelayDays } from '../../data';
+import { useDepositDelayDays, useSettings } from '../../data';
+import Tour from 'components/tour';
 
 const PaymentMethodsDescription = () => (
 	<>
@@ -108,6 +109,96 @@ const DepositsDescription = () => {
 	);
 };
 
+const TourFirstStep = () => {
+	return (
+		<Tour.Step selector="#wpcontent" position={ { bottom: 20, left: 20 } }>
+			<Tour.Image
+				src="https://picsum.photos/350/204"
+				alt="Enhanced fraud protection is here 🔒"
+			/>
+
+			<Tour.Content
+				title="Enhanced fraud protection is here 🔒"
+				// eslint-disable-next-line max-len
+				description="Incoming transactions will now be screened for common risk factors, at the level of your choosing. Review any transactions caught by these filters and select whether you'd like to approve or decline them"
+			/>
+
+			<Tour.Footer showCounter={ false }>
+				<Tour.NextButton />
+			</Tour.Footer>
+		</Tour.Step>
+	);
+};
+
+const TourSecondStep = () => {
+	return (
+		<Tour.Step
+			selector=".deposits__bank-information > h4"
+			position="bottom"
+		>
+			<Tour.Image
+				src="https://picsum.photos/350/204"
+				alt="Choose your filter level 🚦"
+				mobileOnly
+			/>
+
+			<Tour.Content
+				title="Choose your filter level 🚦"
+				// eslint-disable-next-line max-len
+				description="Decide how aggressively you want to filter suspicious payments, from standard to advanced."
+			/>
+
+			<Tour.Footer />
+		</Tour.Step>
+	);
+};
+
+const TourThirdStep = () => {
+	return (
+		<Tour.Step selector=".express-checkout__label" position="left">
+			<Tour.Image
+				src="https://picsum.photos/350/204"
+				alt="Take more control 🎚️"
+				mobileOnly
+			/>
+
+			<Tour.Content
+				title="Take more control 🎚️"
+				// eslint-disable-next-line max-len
+				description="We recommend using one of the preset risk levels, but if you need more control, head to Advanced to fine-tune the various filters."
+			/>
+
+			<Tour.Footer />
+		</Tour.Step>
+	);
+};
+
+const TourFourthStep = () => {
+	return (
+		<Tour.Step
+			selector=".payment-methods__available-methods .payment-method:first-child .payment-method__label"
+			position="right"
+		>
+			<Tour.Image
+				src="https://picsum.photos/350/204"
+				alt="Ready for review 📥️"
+				mobileOnly
+			/>
+
+			<Tour.Content
+				title="Ready for review 📥️"
+				// eslint-disable-next-line max-len
+				description="Payments that have been caught by a risk filter will appear under Transactions > Payments. We'll let you know why each payment was flagged so that you can determine whether to approve or block it."
+			/>
+
+			<Tour.Footer>
+				<Tour.PreviousButton />
+				<Tour.NextButton>Got it!</Tour.NextButton>
+			</Tour.Footer>
+		</Tour.Step>
+	);
+};
+
 const SettingsManager = () => {
 	const {
 		featureFlags: {
@@ -119,6 +210,19 @@ const SettingsManager = () => {
 	const [ isTransactionInputsValid, setTransactionInputsValid ] = useState(
 		true
 	);
+	const { isLoading } = useSettings();
+
+	const [ showTour, setShowTour ] = useState( false );
+
+	useEffect( () => {
+		if ( ! isLoading ) {
+			setShowTour( true );
+		}
+	}, [ isLoading ] );
+
+	const handleTourEnd = () => {
+		setShowTour( false );
+	};
 
 	return (
 		<SettingsLayout>
@@ -176,6 +280,15 @@ const SettingsManager = () => {
 			</SettingsSection>
 			<AdvancedSettings />
 			<SaveSettingsSection disabled={ ! isTransactionInputsValid } />
+
+			{ showTour && (
+				<Tour onTourEnd={ handleTourEnd }>
+					<TourFirstStep />
+					<TourSecondStep />
+					<TourThirdStep />
+					<TourFourthStep />
+				</Tour>
+			) }
 		</SettingsLayout>
 	);
 };
  ```
</details>

-------------------


**Post merge**
